tools/scripts/check_invalid_characters.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# This script does a sanity check for invalid characters in C++ source code
+
+set -e
+
+ICEORYX_PATH=$(git rev-parse --show-toplevel)
+cd $ICEORYX_PATH
+
+function findInFiles() {
+    searchString=$1
+    grep -rn --include=\*.{h,hpp,inl,c,cpp} "\`" iceoryx_*
+}
+
+BACKTICK_SEARCH_STRING="\`"
```

Comment:
@elfenpiff the idea was to just copy&paste that part of the code and add a new invalid character. I was not sure if is is an error if the same variable is defined twice.

If you can confirm that it is not an issue I will change it. But I would not make an array out of it right now since one cannot have a custom error message per character. The complex part is already in a function.

You can of course adjust the script as you like in a follow up PR. This is just an initial minimal viable solution

Reply:
I think this is something for FutureBob or FutureElfe